InnerEye/ML/SSL/datamodules_and_datasets/cxr_datasets.py

```diff
@@ -0,0 +1,155 @@
+    """
+
+    def __init__(self,
+                 root: str,
+                 train: bool,
+                 transform: Optional[Callable] = None,
+                 **kwargs: Any) -> None:
+        """
+
+        :param root: path to the data directory
+        :param train: if True returns the train + val dataset, if False returns the test set. See VisionDataset API.
+        :param transform: callable to be applied on the loaded image, has to take PIL Image as input
+        """
+        super().__init__(root=root, transforms=transform)
+        self.root = Path(self.root)  # type: ignore
+        if not self.root.exists():
+            logging.error(
+                f"The data directory {self.root} does not exist. Make sure to download the data first for the Kaggle "
+                f"page")
+        self.train = train
+        self.targets: Optional[List[int]] = None
+        self._prepare_dataset()
+
+    def _prepare_dataset(self) -> None:
+        self.indices: List[int] = []
+        self.filenames: List[PathOrString] = []
+        raise NotImplementedError("_prepare_dataset needs to be implemented by the child classes.")
+
+    def __getitem__(self, index: int) -> OptionalIndexInputAndLabel:
+        """
+        :param index: The index of the sample to be fetched
+        :return: The image and (fake) label tensors
+        """
+        filename = self.filenames[index]
+        target = self.targets[index] if self.targets is not None else 0
+        if is_dicom_file_path(str(filename)):
+            scan_image = load_dicom_image(filename)
+            # Dicom files have arbitrary pixel intensities, convert to [0,255] range so that PIL can
+            # read the array into an Image object.
+            scan_image = (scan_image - scan_image.min()) * 255. / (scan_image.max() - scan_image.min())
+            # Load as PIL Image in grey-scale (convert("L") step), yields a 1-channel image.
+            scan_image = Image.fromarray(scan_image).convert("L")
+        else:
+            # Load as PIL Image in grey-scale (convert("L") step), yields a 1-channel image with pixel values in range
+            # [0,1] (float).
+            scan_image = Image.open(filename).convert("L")
+        if self.transforms is not None:
+            scan_image = self.transforms(scan_image)
+        return scan_image, target
+
+    def __len__(self) -> int:
+        """
+        :return: The size of the dataset
+        """
+        return len(self.indices)
+
+
+class InnerEyeCXRDatasetWithReturnIndex(InnerEyeDataClassBaseWithReturnIndex, InnerEyeCXRDatasetBase):
+    """
+    Any dataset used in SSL needs to inherit from InnerEyeDataClassBaseWithReturnIndex as well as VisionData.
+    This class is just a shorthand notation for this double inheritance.
+    """
+    pass
+
+
+class RSNAKaggleCXR(InnerEyeCXRDatasetWithReturnIndex):
+    """
+    Dataset class to load the RSNA Chest-Xray training dataset. Use all the data for train and val. No test data
```

Comment:
Can we write down how the files should be placed in a blob-storage or local drive?

what is `dataset.csv`?
where to find the dataset? https://www.kaggle.com/c/rsna-pneumonia-detection-challenge/data

Reply:
Pointed the user to the docs with the instructions as you suggested below